relayer/metrics/metricsService.go

```diff
@@ -0,0 +1,175 @@
+package metrics
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/lavanet/lava/utils"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+type AggregatedMetric struct {
+	TotalLatency int64
+	RelaysCount  uint64
+	SuccessCount int64
+}
+
+type MetricService struct {
+	numberOfRelays      int64
+	aggregatedMetricMap *map[string]map[string]map[string]*AggregatedMetric
+	metricsChannel      chan RelayAnalytics
+	sqsUrl              string
+}
+
+func NewMetricService() *MetricService {
+	metricPortalSqsUrl := os.Getenv("METRICS_PORTAL_SQS_URL")
+	intervalData := os.Getenv("METRICS_INTERVAL_FOR_SENDING_DATA_INM")
+	if metricPortalSqsUrl == "" || intervalData == "" {
+		return nil
+	}
+	intervalForMetrics, _ := strconv.ParseInt(intervalData, 10, 32)
+	utils.LavaFormatDebug("env variables:", &map[string]string{
+		"url":      metricPortalSqsUrl,
+		"interval": intervalData})
+	mChannel := make(chan RelayAnalytics)
+	result := &MetricService{
+		metricsChannel:      mChannel,
+		sqsUrl:              metricPortalSqsUrl,
+		aggregatedMetricMap: &map[string]map[string]map[string]*AggregatedMetric{},
+	}
+	//setup reader
+	go result.ReadFromChannel()
+	//setup sending of the results into the query
+	ticker := time.NewTicker(time.Duration(intervalForMetrics * time.Minute.Nanoseconds()))
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
```

Comment:
we should have here another case for reading from the channel.
so when it ticks we don't read, and when it reads from the channel we don't send data

Reply:
see last example here 
https://dev.to/julianchu/go-for-select-with-timer-1ah8

Reply:
right now you have a race condition between adding metrics and reseting the map